server/src/main/java/org/opensearch/tasks/TaskManager.java

```diff
@@ -138,17 +139,21 @@ public static TaskManager createTaskManagerWithClusterSettings(
         ThreadPool threadPool,
         Set<String> taskHeaders
     ) {
-        final TaskManager taskManager = new TaskManager(settings, threadPool, taskHeaders);
+        final TaskManager taskManager = new TaskManager(settings, threadPool, taskHeaders, clusterSettings);
         clusterSettings.addSettingsUpdateConsumer(TASK_RESOURCE_CONSUMERS_ENABLED, taskManager::setTaskResourceConsumersEnabled);
         return taskManager;
     }
 
     public TaskManager(Settings settings, ThreadPool threadPool, Set<String> taskHeaders) {
+        this(settings, threadPool, taskHeaders, null);
+    }
+
+    private TaskManager(Settings settings, ThreadPool threadPool, Set<String> taskHeaders, @Nullable ClusterSettings clusterSettings) {
         this.threadPool = threadPool;
         this.taskHeaders = new ArrayList<>(taskHeaders);
         this.maxHeaderSize = SETTING_HTTP_MAX_HEADER_SIZE.get(settings);
         this.taskResourceConsumersEnabled = TASK_RESOURCE_CONSUMERS_ENABLED.get(settings);
-        this.taskResourceConsumer = Set.of(new TopNSearchTasksLogger(settings));
+        this.taskResourceConsumer = Set.of(new TopNSearchTasksLogger(settings, clusterSettings));
```

Comment:
Can we pass a collection of task consumers as a dependency to the `TaskManager`? Or expose a method to register a task consumer in order to decouple dependencies.